libarchive/archive_read_support_format_zip.c

```diff
@@ -3433,6 +3434,8 @@ read_eocd(struct zip *zip, const char *p, int64_t current_offset)
 
 	/* Save the central directory location for later use. */
 	zip->central_directory_offset = archive_le32dec(p + 16);
+	zip->central_directory_offset_actual = current_offset
+		- archive_le32dec(p+12) - 1;
```

Comment:
Why is there a `- 1` here?  (Might be worth a comment.)

Not your fault, but this whole function would be easier to read if it gave names to all the values here:
```
   uint64_t central_directory_size = archive_le32dec(p + 12);
   zip->central_directory_offset_actual = current_offset - central_directory_size - 1;
```